engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMapBuilder.java

```diff
@@ -1,193 +1,264 @@
 package org.enso.interpreter.runtime.data.hash;
 
-import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
+
 import org.enso.interpreter.node.expression.builtin.meta.EqualsNode;
 import org.enso.interpreter.node.expression.builtin.meta.HashCodeNode;
-import org.graalvm.collections.EconomicMap;
-import org.graalvm.collections.Equivalence;
+
+import com.oracle.truffle.api.CompilerDirectives;
 
 /**
- * A storage for a {@link EnsoHashMap}. For one builder, there may be many snapshots ({@link
- * EnsoHashMap}). There should be at most one snapshot for a given size. All the snapshots should
- * have size smaller than this builder size.
+ * A storage for a {@link EnsoHashMap}. For one builder, there may be many
+ * {@link EnsoHashMap} instances that serve as a snapshot.
+ *
+ * There should be where most one snapshot for a given generation. All the snapshots should
```

Comment:
This sentence does not make sense. Should it be: "There should be at most one snapshot for a given generation"?

Reply:
Done in 6522d31055